### PR TITLE
Here are a few fixes utils.network

### PR DESCRIPTION
**Pull request by sakateka**
- not use lsof if proc is available
- support invert options of hex2ip for ipv6
- not issue an error if lsof does not find the TCP sessions

### CODE REVIEW

File: salt/utils/network.py [hunk 1]
```diff
@@ -973,14 +973,17 @@ def ip_addrs6(interface=None, include_loopback=False, interface_data=None):
...
+    returned. If 'invert=True' assume that ip from /proc/net/<proto>
     '''
     if len(hex_ip) == 32:  # ipv6
         ip = []
         for i in range(0, 32, 8):
             ip_part = hex_ip[i:i + 8]
             ip_part = [ip_part[x:x + 2] for x in range(0, 8, 2)]
-            ip.append("{0[3]}{0[2]}:{0[1]}{0[0]}".format(ip_part))
+            if invert:
+                ip.append("{0[3]}{0[2]}:{0[1]}{0[0]}".format(ip_part))
```
Comment (cachedout) [inline salt/utils/network.py]: Given that the default for `invert` is `False`, doesn't this imply a change in behavior here? Is this a bug fix or was this unintentional?

**Comment by cachedout** [comment]
Hi @sakateka I just had on question in here that I asked in-line to the PR. If you could take a look and just help ease my mind, we can go from there. Thanks!

File: salt/utils/network.py (hunk 1 above)
Comment (sakateka) [inline salt/utils/network.py]: Hi!
I looked at all of the code and all calls made with hex2ip ```invert=True```, these changes simply retain the previous agreement of the call hex2ip.
I do not know the original intentions of the author of the ```hex2ip``` but it is my opinion that the setting invert can be removed together with the excess code.

Comment (cachedout) [inline salt/utils/network.py]: Sounds good. Just wanted to make sure we covered our bases there. Thanks for the reply.

**Pull request merged**